### ISSUE TITLE
Reference to the content

### ISSUE THREAD

**Issue by srib** [Forgotten/jax_hps#1]
Is the implementation based on [this](https://arxiv.org/pdf/1506.01308.pdf) paper?

**Comment by Forgotten** [comment]
yes, it implements algorihtm 4 in that paper.  Although the implementation is very rudimentary. It was mostly designed to understand the limitations of JAX when using nested data structures.
